src/shared/utilities/timeoutUtils.ts

```diff
@@ -118,3 +123,41 @@ export async function waitUntil<T>(
         await new Promise(r => setTimeout(r, opt.interval))
     }
 }
+
+/**
+ * Utility function to wrap a Timeout token around a Promise.
+ *
+ * @param promise The promise to use a Timeout with
+ * @param timeout A Timeout token that will race against the promise
```

Comment:
Articles ("A ...") at the start of a docstring are usually noise, which diminishes the useful part of the description.
```suggestion
 * @param timeout Timeout token that will race against the promise
```